src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1387,17 +1403,27 @@ private function dumpValue($value, $interpolate = true)
 
             return sprintf('array(%s)', implode(', ', $code));
         } elseif ($value instanceof IteratorArgument) {
+            $countCode = array();
+            $countCode[] = 'function() {';
+            $operands = array(0);
+
             $code = array();
             $code[] = 'new RewindableGenerator(function() {';
-            foreach ($value->getValues() as $k => $v) {
```

Comment:
`$v` is nice, but a need to exploit `$value` again for conditionals count. `$v` is overwritten by:

```php
$v = $this->wrapServiceConditionals($value, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($value,
```

Can use something else for this though.